unblob/finder.py

```diff
@@ -19,8 +28,68 @@
 """
 
 
+def search_chunks_by_priority(  # noqa: C901
+    path: Path, file: io.BufferedReader, file_size: int
+) -> List[ValidChunk]:
+    """Search all ValidChunks within the file.
+    Collect all the registered handlers by priority, search for YARA patterns and run
+    Handler.calculate_chunk() on the found matches.
+    We don't deal with offset within already found ValidChunks and invalid chunks are thrown away.
+    """
+    all_chunks = []
+
+    for priority_level, handler_classes in enumerate(ALL_HANDLERS_BY_PRIORITY, start=1):
+        logger.info("Starting priority level", priority_level=noformat(priority_level))
+        yara_rules = make_yara_rules(handler_classes)
+        handler_map = make_handler_map(handler_classes)
+        yara_results = search_yara_patterns(yara_rules, handler_map, path)
+
+        for result in yara_results:
+            handler, match = result.handler, result.match
+
+            sorted_matches = sorted(match.strings, key=itemgetter(0))
+            for offset, identifier, string_data in sorted_matches:
+                real_offset = offset + handler.YARA_MATCH_OFFSET
+
+                if any(chunk.contains_offset(real_offset) for chunk in all_chunks):
```

Comment:
What does this mean? What is the decision we are making here. In other words: what does the loop's body do below this line?

Reply:
Added a comment about this.